relayer/src/services/types.ts

```diff
@@ -106,7 +143,88 @@ export abstract class BaseRelayerService {
     return;
   }
 
-  protected async parseFillEventDataFromTxHash(
+  async checkSuccessOnRequestManager(
+    resolvedRequestId: string,
+    resolvedFillId: string,
+    timeoutMs = 300000,
+  ): Promise<boolean> {
+    const resolverAddress = L1_CONTRACTS[this.l1ChainId]?.RESOLVER ?? process.env.RESOLVER;
+    const resolver = Resolver__factory.connect(resolverAddress, this.l1RpcProvider);
+    const requestManagerAddress = (await resolver.sourceChainInfos(this.l2ChainId)).requestManager;
+    const requestManager = RequestManager__factory.connect(
+      requestManagerAddress,
+      this.l2RpcProvider,
+    );
+
+    const timeout = new Promise<undefined>((resolve) => setTimeout(resolve, timeoutMs)).then(
+      () => {
+        requestManager.removeAllListeners();
```

Comment:
Nitpick: I think we can move this additional clean-up step inside a function that will be provided to the `setTimeout()` call (where the call to `resolve` will happen too).